src/go/rpk/pkg/cli/security/role/role.go

```diff
@@ -0,0 +1,65 @@
+// Copyright 2024 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+package role
+
+import (
+	"strings"
+
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/adminapi"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/config"
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+)
+
+const (
+	rolePrefix = "RedpandaRole:"
+	userPrefix = "User:"
+)
+
+func NewCommand(fs afero.Fs, p *config.Params) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "role",
+		Aliases: []string{"access", "roles"},
+		Args:    cobra.ExactArgs(0),
+		Short:   "Manage Redpanda roles",
+	}
+	cmd.AddCommand(
+		assignCommand(fs, p),
+		createCommand(fs, p),
+		deleteCommand(fs, p),
+		describeCommand(fs, p),
+		listCommand(fs, p),
+		unassignCommand(fs, p),
+	)
+	p.InstallFormatFlag(cmd)
+	return cmd
+}
+
+// parsePrincipal returns the prefix, and principal. If no prefix is present,
+// returns User:
+func parsePrincipal(p string) (principalType string, name string) {
+	if strings.HasPrefix(p, userPrefix) {
+		return "User", strings.TrimPrefix(p, userPrefix)
+	}
+	if strings.HasPrefix(p, rolePrefix) {
+		return "RedpandaRole", strings.TrimPrefix(p, rolePrefix)
+	}
+	return "User", p
+}
+
+// parseRoleMembers parses a --principal flag to a []adminapi.RoleMember
+func parseRoleMember(principals []string) []adminapi.RoleMember {
+	var members []adminapi.RoleMember
+	for _, p := range principals {
+		pType, name := parsePrincipal(p)
```

Comment:
worth noting that roles can't be role members. is that captured here? worst case is that a request would fail, but you could disallow it at the interface if so inclined.

Reply:
Yes, we have the logic to support it, and Redpanda fails if it gets passed... but I will change it to an error in rpk instead in the follow-up ducktape test PR

Reply:
cool, nbd from my perspective

Reply:
Is being [removed here](https://github.com/redpanda-data/redpanda/pull/17617/commits/bd255fb5960a2b3cccefd3d487604dfd6cfe8eea#diff-c762328733c900efc31d260ac87625869915e6371a3f95cb8d4d28c3923c6044L51) in the PR #17617

Thanks :+1: